### PR TITLE
8304712: Only pass total number of regions into G1Policy::calc_min_old_cset_length

### PR DESCRIPTION
**Pull request by tschatzl**
Hi all,

  please review this change that refactors `G1Policy::calc_min_old_cset_length` to take only required parameters and also just use `ceil()` for the calculation. It just does not matter to be clever here imo.

Testing: gha

Thanks,
  Thomas

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8304712](https://bugs.openjdk.org/browse/JDK-8304712): Only pass total number of regions into G1Policy::calc_min_old_cset_length

### Reviewers
 * [Albert Mingkun Yang](https://openjdk.org/census#ayang) (@albertnetymk - **Reviewer**) ⚠️ Review applies to [13c093c6](https://git.openjdk.org/jdk/pull/13147/files/13c093c6a1683af189bd1d2a1a3be179a850308d)
 * [Ivan Walulya](https://openjdk.org/census#iwalulya) (@walulyai - **Reviewer**) ⚠️ Review applies to [13c093c6](https://git.openjdk.org/jdk/pull/13147/files/13c093c6a1683af189bd1d2a1a3be179a850308d)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/13147/head:pull/13147` \
`$ git checkout pull/13147`

Update a local copy of the PR: \
`$ git checkout pull/13147` \
`$ git pull https://git.openjdk.org/jdk.git pull/13147/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 13147`

View PR using the GUI difftool: \
`$ git pr show -t 13147`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/13147.diff">https://git.openjdk.org/jdk/pull/13147.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/share/gc/g1/g1Policy.cpp [hunk 1]
```diff
@@ -1380,7 +1380,7 @@ size_t G1Policy::allowed_waste_in_collection_set() const {
   return G1HeapWastePercent * _g1h->capacity() / 100;
 }
 
-uint G1Policy::calc_min_old_cset_length(G1CollectionSetCandidates* candidates) const {
+uint G1Policy::calc_min_old_cset_length(uint num_marked_regions) const {
```
Comment (albertnetymk) [inline src/hotspot/share/gc/g1/g1Policy.cpp:1383]: I actually think the previous signature is easier to read; The name, `num_marked_regions`, doesn't tell me it's in fact the number of candidates.

**Review by albertnetymk — APPROVED** [review]

**Review by walulyai — APPROVED** [review]

File: src/hotspot/share/gc/g1/g1Policy.cpp (hunk 1 above)
Comment (tschatzl) [reply src/hotspot/share/gc/g1/g1Policy.cpp:1383]: Later there will be a distinction within the candidates between regions added by marking, and regions added by evacuation failure/pinning. This is a very important distinction for this method to keep mixed gcs working as they are. The comment explains it. I would prefer to not change the signature again a few PRs later.

As for the previous signature being easier to read, I think keeping the information passed to method as succinct as possible makes it easier to reason about what the method actually does (and needs), so I would like to keep it as is. We can certainly argue about the actual naming (`num_marked_regions` needs a bit of an explanation as done in the comment), but I could not find something better.
I.e. `f(int) -> int` is imo easier than `f(class with lots of stuff inside) -> int`.

Comment (albertnetymk) [reply src/hotspot/share/gc/g1/g1Policy.cpp]: > Later there will be a distinction

I think one can cross that bridge when we get there. Given limited info in this PR, it's confusing to highlight the diff prematurely, IMO.

**Comment by tschatzl** [comment]
Thanks for your reviews

/integrate

**Pull request closed without merging**